### PR TITLE
Validate git based ruby gems work with bundler source

### PR DESCRIPTION
**Pull request by jonabc**
closes https://github.com/github/licensed/issues/359

The issue here is that the path construction for the search root results in a path that doesn't exist and is not an ancestor for gems pulled via a "git" directive in bundlers gemfile.  The change in this PR ensures that the root path exists before it's returned.  This should be ok for the case given but doesn't fully address the use case for git sourced bundler dependencies and may need to be further addressed in the future.  For now, this is a small change that is good to have regardless.

I validated this change originally using the `gem` line shared in the related issue, but opted to use a smaller gem that pulls in fewer dependencies and is not a fork in the checked in test fixture

cc clinejj

### CODE REVIEW

**Comment by jonabc** [comment]
I'm not seeing any of these errors in my local environment, digging into that

**Comment by jonabc** [comment]
TL;DR - I'm not sure why I added search_root usage to the bundler source but it appears to not be needed.  Since it was causing issues I removed it.

After doing some debugging both locally and logging values in CI I don't see what value the search_root argument is bringing to bundler dependencies.  Generally when gems get installed they get their own folder where the folder name includes both the name and the version.  Additionally, each gem is packaged and installed as a standalone unit where installed gems don't normally share files with each other.

The search_root argument was added to support the `go` source, where multiple go package imports can come from the same repository and share a LICENSE file at the root of the repo's codebase rather than in each subfolder.  That situation doesn't apply here though and IIRC I used the search_root value to resolve a test failure that no longer seems to be needed?

🤷
